docs/design/HealthEndpoint.md

```diff
@@ -0,0 +1,417 @@
+| `max-dop`     | Integer       | No           | `8`         | Maximum Degree of Parallelism for running health checks.                                             |
+| `roles`       | Array of strings         | <table><tr><th>Environment</th><th>Required</th></tr><tr><td>Development</td><td>No</td></tr><tr><td>Production</td><td>Yes</td></tr></table>   | <table><tr><th>Environment</th><th>Default</th></tr><tr><td>Development</td><td>["anonymous"]</td></tr><tr><td>Production</td><td>NA</td></tr></table>        | List of Roles allowed to access the comprehensive health endpoint (e.g., `anonymous`, `authenticated` or `customer-role`).                   |
+
+```
+{
+  "runtime": {
+    "health": {
+      "enabled": true,
+      "cache-ttl-seconds": 5,
+      "max-dop": 5,
+      "roles": ["anonymous", "authenticated"]
+    }
+  }
+}
+```
+
+#### `data-source.health` Configuration
+
+The database type in the data source health config determines the query that we should run on the data source. The time to run this query should come under the user specified threshold_ms to qualify as a healthy data source for DAB. We get the database type from the runtime parameters to get the query to run on the specific DB Type.\
+**This check opens a connection to the data source and validates the connection string and network access. Further it runs a simple query to validate user permissions and responsiveness or the DB.**
+
+| **Property**      | **Data Type** | **Required** | **Default** | **Description**                                                                                      |
+|-------------------|---------------|--------------|-------------|------------------------------------------------------------------------------------------------------|
+| `enabled`         | Boolean       | No           | `true`      | Enables or disables health checks for the data source.                                             |
+| `name`         | String        | No           | Database Type `runtime.data-source.database-type`     | Identifier for the data source. Useful when multiple data sources exist, to identify each independently in the health check.  
+| `threshold-ms`    | Integer       | No           | `10000`     | Threshold in milliseconds for the query response time before the check is considered unhealthy. Minimum : 1 Maximum: INT_MAX. Zero is not allowed.      |
+
+```
+{
+  "data-source": {
+    "health": {
+      "name": "sqlserver",
+      "enabled": true,
+      "threshold-ms": 100
+    }
+  }
+}
+```
+
+#### `<entity-name>.health` Configuration
+
+The Entity config parameters contain information about the `first` which defines the number of first rows to be returned when executing a READ operation on the data source entity and under what threshold ms should the response be received for it to qualify as a healthy entity.
```

Comment:
nit: parameters > settings
nit: defines the number of first rows > defines the number of rows
nit: under what threshold ms should the response be received > under what threshold the response should be received